app/controllers/api_controller/collection_config.rb

```diff
@@ -0,0 +1,110 @@
+class ApiController
+  class CollectionConfig < Config::Options
+    def collection?(collection_name)
+      c(collection_name)[:options].include?(:collection)
+    end
+
+    def custom_actions?(collection_name)
+      cspec = c(collection_name)
+      cspec && cspec[:options].include?(:custom_actions)
+    end
+
+    def primary?(collection_name)
+      c(collection_name)[:options].include?(:primary)
+    end
+
+    def show?(collection_name)
+      c(collection_name)[:options].include?(:show)
+    end
+
+    def show_as_collection?(collection_name)
+      c(collection_name)[:options].include?(:show_as_collection)
+    end
+
+    def supports_http_method?(collection_name, method)
+      Array(c(collection_name)[:verbs]).include?(method)
+    end
+
+    def subcollections(collection_name)
+      Array(c(collection_name)[:subcollections])
+    end
+
+    def subcollection?(collection_name, subcollection_name)
+      subcollections(collection_name).include?(subcollection_name.to_sym)
+    end
+
+    def subcollection_denied?(collection_name, subcollection_name)
+      c(collection_name)[:subcollections] && !c(collection_name)[:subcollections].include?(subcollection_name.to_sym)
+    end
+
+    def typed_collection_actions(collection_name, target)
+      c(collection_name)["#{target}_actions".to_sym]
+    end
+
+    def typed_subcollection_actions(collection_name, subcollection_name)
+      c(collection_name)["#{subcollection_name}_subcollection_actions".to_sym]
+    end
+
+    def typed_subcollection_action(collection_name, subcollection_name, method)
+      typed_subcollection_actions(collection_name, subcollection_name).try(:fetch_path, method.to_sym)
+    end
+
+    def local_uris_for_feature(product_feature_name)
```

Comment:
I'm seeing this more clearly now :) Maybe this one (and the private method associated with it) could be named so that it reflects that it's returning the collection names for that feature? And have a method somewhere that does the building of the URI?